### PR TITLE
VAULT-6727 Add role resolution for CF

### PR DESCRIPTION
**Pull request by VioletHynes**
Support role resolution functionality (to work with role based quotas). The intent of this operation is to take the same data as a login request and return the role that would be used in the request, so that we can quota upon login requests that have a certain role.

Afterwards, I will push a new tag and pull this new version into Vault.

### CODE REVIEW

**Review by ccapurso — APPROVED** [review]
Looks good!

**Review by hghaf099 — APPROVED** [review]
Couple nit comments, otherwise looks good

File: path_login.go [hunk 1]
```diff
@@ -61,12 +61,34 @@ func (b *backend) pathLogin() *framework.Path {
...
 		HelpDescription: pathLoginDesc,
 	}
 }
 
+// resolveRole resolves the role that will be used from this login request.
+// This is used as part of role-based quotas.
+func (b *backend) resolveRole(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	roleName := data.Get("role").(string)
+	if roleName == "" {
+		return logical.ErrorResponse("'role-name' is required"), nil
```
[the remark this answers is not in the document]
Comment (VioletHynes) [reply path_login.go:78]: This error response is kept consistent with the login endpoint, and I would be wary about changing the login endpoint response

File: path_login.go [hunk 2]
```diff
@@ -61,12 +61,34 @@ func (b *backend) pathLogin() *framework.Path {
...
+				Callback: b.resolveRole,
+			},
 		},
 		HelpSynopsis:    pathLoginSyn,
 		HelpDescription: pathLoginDesc,
 	}
 }
 
+// resolveRole resolves the role that will be used from this login request.
+// This is used as part of role-based quotas.
```
[the remark this answers is not in the document]
Comment (VioletHynes) [reply path_login.go]: Good point, fixed

**Review by tomhjp — APPROVED** [review]
LGTM

**Pull request merged**